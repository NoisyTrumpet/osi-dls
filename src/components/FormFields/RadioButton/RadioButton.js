import React from 'react';
import PropTypes from 'prop-types';
import { useField } from 'formik';
import classNames from 'classnames';
import { Label } from '../Label';
import { Input } from '../Input';
import { Indicator } from '../Indicator';
import { HelperText } from '../HelperText';
import './RadioButton.scss';

const RadioButton = ({
  ariaControls,
  domId,
  label,
  isCard,
  helperText,
  addClass,
  isDisabled,
  children,
  variant,
  hiddenLabel,
  ...props
}) => {
  // useField passes name, placeholder, value, onChange and onBlur
  const [field] = useField({ ...props, type: 'radio' });

  // For FieldSet Render Props
  const { name } = props;

  const classes = classNames(
    'input-field-container',
    'input-field-radio',
    { 'input-field-radio-blank': variant === 'blank-card' },
    { 'input-field-radio-card': isCard },
    { 'input-field-disabled': isDisabled },
    addClass,
  );

  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <div className={classes}>
      <Input
        ariaControls={ariaControls}
        type="radio"
        domId={domId}
        name={name || field.name}
        isChecked={field.checked}
        className={classNames('radio-button')}
        isDisabled={isDisabled}
        // Spreading props, as required by Formik in useField: https://jaredpalmer.com/formik/docs/api/useField#example
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...field}
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...props}
      />
      <Indicator type={variant === 'blank-card' ? 'hidden' : 'radio'} />
      {children}
      <div className={`radio-button-labels ${hiddenLabel && 'screen-reader-text'}`}>
        <Label
          fieldType="radio"
          isDisabled={isDisabled}
          forId={domId}
          display="inline-block"
          addClass={hiddenLabel && "screen-reader-text"}
          required
        >
          {label}
        </Label>
        {isCard && helperText ? (
          <HelperText>{helperText}</HelperText>
        ) : ''}
      </div>
    </div>
  );
};

RadioButton.defaultProps = {
  isCard: false,
  isDisabled: false,
};

RadioButton.propTypes = {
  ariaControls: PropTypes.string,
  label: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
  ]),
  hiddenLabel: PropTypes.bool,
  name: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.number,
    PropTypes.string,
  ]),
  domId: PropTypes.string.isRequired,
  addClass: PropTypes.string,
  helperText: PropTypes.string,
  isCard: PropTypes.bool,
  isDisabled: PropTypes.bool,
  children: PropTypes.node,
  variant: PropTypes.oneOf([
    'blank-card',
  ])
};

export default RadioButton;
