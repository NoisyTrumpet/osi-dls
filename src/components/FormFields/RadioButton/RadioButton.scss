.input-field-radio {
  background: map-get($colors, 'brand-white' );
  
  &.input-field-disabled,
  .input-label.isDisabled {
    .typography {
      color: map-get($colors, 'gray3');
    }

    input[type=radio] {
      border: rem(1px) solid map-get($colors, 'gray2');
    }
  }
  .indicator {
    align-self: flex-start;
    margin-top: rem(3px);
    min-width: rem(18px);
  }
}

.input-field-radio-card {
  padding: rem(12px);
  margin: rem(16px) auto;
  border-radius: rem(7px);

  // this will make the whole card clickable and toggle the radio button
  input[type=radio] {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    opacity: 1;
    border: rem(1px) solid map-get($colors, 'gray3');

    &:hover {
      border: rem(1px) solid map-get($colors, 'brand-blue');
    }

    &:checked {
      border: rem(2px) solid map-get($colors, 'brand-blue');
    }
  }

  .indicator {
    align-self: flex-start;
    margin-top: rem(3px);

    &:after {
      top: initial;
      left: initial;
    }
  }

  > div {
    width: 90%;

    label.input-label {
      display: block;
    }
  }

  .helper-text {
    margin-top: 0;
    line-height: 1.125;
  }

  &.input-field-disabled {
    .typography {
      color: map-get($colors, 'disabled');
    }

    input[type=radio] {
      border: rem(1px) solid map-get($colors, 'disabled');
    }
  }
}

.input-field-radio-blank {
  padding: rem(16px);

  input[type=radio] {
    border: 0;
    box-shadow: $shadow-1;

    &:hover {
      border: 0;
      box-shadow: $shadow-2;
    }

    &:checked {
      border: 0;
      box-shadow: inset $shadow-0;
    }

    &:focus {
      outline: rem(1px) dotted #212121;
      outline: rem(1px) auto -webkit-focus-ring-color;
    }
  }

  .radio-button-labels {
    @extend %screen-reader-text;
  }
}
