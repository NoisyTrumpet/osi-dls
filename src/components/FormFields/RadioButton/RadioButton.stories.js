import React from "react";
import { storiesOf } from "@storybook/react";
import withFormik from "storybook-formik";
import { RadioButton } from "./";
import { Image } from "../../Image";
import { Button } from "../../Button";

const storyTitle = "components | Form Fields/Radio Button"; // Stories can be organized in a nested structure using ”/” as a separator, and can be given a top-level heading using a ”|” root separator

const RadioButtonParameters = {
  component: RadioButton,
  componentSubtitle: "Utilize <RadioButton /> to present radio buttons",
  jest: ["RadioButton"]
};

const imageArray = [
  <div style={{ marginRight: "16px" }} key="1">
    <Image
      className="radio-button-image"
      alt="pic-1"
      src="https://media4.s-nbcnews.com/j/streams/2014/May/140505/2D274905781707-hallmark-two-moms-day-md2926.fit-760w.jpg"
    />
  </div>,
  <div style={{ marginRight: "16px" }} key="2">
    <Image
      className="radio-button-image"
      alt="pic-2"
      src="https://media4.s-nbcnews.com/j/streams/2014/May/140505/2D274905781707-hallmark-two-moms-day-md2926.fit-760w.jpg"
    />
  </div>,
  <div key="3">
    <Image
      className="radio-button-image"
      alt="pic-3"
      src="https://media4.s-nbcnews.com/j/streams/2014/May/140505/2D274905781707-hallmark-two-moms-day-md2926.fit-760w.jpg"
    />
  </div>
];

storiesOf(storyTitle, module)
  .addParameters(RadioButtonParameters)
  .addDecorator(withFormik)
  .add("Default", () => (
    <div>
      <RadioButton
        label="Radio Button 1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        checked
      />
      <RadioButton
        label="Radio Button 2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
      />
    </div>
  ))
  .add("Disabled", () => (
    <div>
      <RadioButton
        label="Radio Button 1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        checked
        isDisabled
      />
      <RadioButton
        label="Radio Button 2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
        isDisabled
      />
    </div>
  ))
  .add("Cards", () => (
    <div>
      <RadioButton
        label="Radio Button Card 1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        isCard
        helperText="Optional helper text"
        checked
      />
      <RadioButton
        label="Radio Button Card 2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
        isCard
        helperText="Optional helper text"
      />
      <RadioButton
        label="Radio Button Card 3"
        name="myRadioGroup"
        domId="myDomId3"
        value="Option 3"
        isCard
        helperText="Optional helper text"
        isDisabled
      />
    </div>
  ))
  .add("Blank Cards", () => (
    <div style={{ width: "400px" }}>
      <RadioButton
        variant="blank-card"
        label="image1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        isCard
        checked
      >
        {imageArray}
      </RadioButton>
      <RadioButton
        variant="blank-card"
        label="image2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
        isCard
      >
        {imageArray}
      </RadioButton>
      <RadioButton
        variant="blank-card"
        label="image3"
        name="myRadioGroup"
        domId="myDomId3"
        value="Option 3"
        isCard
      >
        Can be text or any other type of content
      </RadioButton>
    </div>
  ))
  .add("Cards contained in a 250px wide <div>", () => (
    <div style={{ width: 250 }}>
      <RadioButton
        label="Radio Button Card 1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        isCard
        helperText="Optional helper text"
        checked
      />
      <RadioButton
        label="Radio Button Card 2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
        isCard
        helperText="Optional helper text"
      />
      <RadioButton
        label="Radio Button Card 3"
        name="myRadioGroup"
        domId="myDomId3"
        value="Option 3"
        isCard
        helperText="Optional helper text"
        isDisabled
      />
    </div>
  ))
  .add("Regular Radio With Child", () => (
    <div>
      <RadioButton
        label="Radio Button 1"
        name="myRadioGroup"
        domId="myDomId"
        value="Option 1"
        checked
      >
        <p style={{ paddingLeft: "20px" }}>
          This can be any kind of content, needs to be styled by child
        </p>
      </RadioButton>
      <RadioButton
        label="Radio Button 2"
        name="myRadioGroup"
        domId="myDomId2"
        value="Option 2"
      >
        <Button disabled={false} iconLeft="arrow">
          Even a Button
        </Button>
      </RadioButton>
    </div>
  ));
