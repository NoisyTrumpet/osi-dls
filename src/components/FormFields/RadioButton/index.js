export { default as RadioButton } from './RadioButton'; // eslint-disable-line import/prefer-default-export
