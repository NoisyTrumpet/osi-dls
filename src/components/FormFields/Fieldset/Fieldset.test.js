import React from 'react';
import { mount } from 'enzyme';
import { Formik, Form } from 'formik';
import toJson from 'enzyme-to-json';
import { RadioButton } from '../RadioButton';
import { Fieldset } from './';

describe('<Fieldset />', () => {
  it('Should render Radio Buttons', () => {
    const wrapper = mount(
      <Formik initialValues={{ radioGroup: '' }}>
        <Form>
          <Fieldset
            label="RadioButton"
            domId="radioGroup"
            helperText="helperText"
            name="radioGroup"
          >
            <RadioButton
              label="Radio Btn 1"
              name="radioGroup"
              type="radio"
              value="radioBtn1"
              domId="radioBtn1"
            />
            <RadioButton
              label="Radio Btn 2"
              name="radioGroup"
              type="radio"
              value="radioBtn2"
              domId="radioBtn2"
            />
          </Fieldset>
        </Form>
      </Formik>,
    );
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
