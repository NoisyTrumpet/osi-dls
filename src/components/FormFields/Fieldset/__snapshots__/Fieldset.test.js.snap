// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> Should render Radio Buttons 1`] = `
<Formik
  initialValues={
    Object {
      "radioGroup": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <Fieldset
        domId="radioGroup"
        helperText="helperText"
        label="RadioButton"
        name="radioGroup"
      >
        <div
          className="input-fieldset fieldset-container"
        >
          <fieldset
            id="radioGroup"
            label="RadioButton"
            name="radioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <legend
              className="screen-reader-text"
            />
            <RadioButton
              domId="radioBtn1"
              isCard={false}
              isDisabled={false}
              label="Radio Btn 1"
              name="radioGroup"
              type="radio"
              value="radioBtn1"
            >
              <div
                className="input-field-container input-field-radio"
              >
                <Input
                  checked={false}
                  className="radio-button"
                  domId="radioBtn1"
                  isChecked={false}
                  isDisabled={false}
                  name="radioGroup"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder=""
                  type="radio"
                  value="radioBtn1"
                >
                  <input
                    checked={false}
                    className="input-field"
                    defaultValue="radioBtn1"
                    disabled={false}
                    id="radioBtn1"
                    name="radioGroup"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder=""
                    type="radio"
                  />
                </Input>
                <Indicator
                  type="radio"
                >
                  <span
                    aria-hidden="true"
                    className="indicator indicator-radio"
                  />
                </Indicator>
                <div
                  className="radio-button-labels undefined"
                >
                  <Label
                    color=""
                    display="inline-block"
                    fieldType="radio"
                    forId="radioBtn1"
                    isDisabled={false}
                    required={true}
                  >
                    <label
                      className="input-label radio inline-block"
                      htmlFor="radioBtn1"
                    >
                      <Typography
                        color=""
                        variant="field-label"
                      >
                        <span
                          className="typography typography-field-label"
                        >
                          Radio Btn 1
                        </span>
                      </Typography>
                    </label>
                  </Label>
                </div>
              </div>
            </RadioButton>
            <RadioButton
              domId="radioBtn2"
              isCard={false}
              isDisabled={false}
              label="Radio Btn 2"
              name="radioGroup"
              type="radio"
              value="radioBtn2"
            >
              <div
                className="input-field-container input-field-radio"
              >
                <Input
                  checked={false}
                  className="radio-button"
                  domId="radioBtn2"
                  isChecked={false}
                  isDisabled={false}
                  name="radioGroup"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder=""
                  type="radio"
                  value="radioBtn2"
                >
                  <input
                    checked={false}
                    className="input-field"
                    defaultValue="radioBtn2"
                    disabled={false}
                    id="radioBtn2"
                    name="radioGroup"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder=""
                    type="radio"
                  />
                </Input>
                <Indicator
                  type="radio"
                >
                  <span
                    aria-hidden="true"
                    className="indicator indicator-radio"
                  />
                </Indicator>
                <div
                  className="radio-button-labels undefined"
                >
                  <Label
                    color=""
                    display="inline-block"
                    fieldType="radio"
                    forId="radioBtn2"
                    isDisabled={false}
                    required={true}
                  >
                    <label
                      className="input-label radio inline-block"
                      htmlFor="radioBtn2"
                    >
                      <Typography
                        color=""
                        variant="field-label"
                      >
                        <span
                          className="typography typography-field-label"
                        >
                          Radio Btn 2
                        </span>
                      </Typography>
                    </label>
                  </Label>
                </div>
              </div>
            </RadioButton>
          </fieldset>
          <HelperText>
            <div
              className="helper-text"
            >
              <Typography
                variant="helper"
              >
                <span
                  className="typography typography-helper"
                >
                  helperText
                </span>
              </Typography>
            </div>
          </HelperText>
        </div>
      </Fieldset>
    </form>
  </Form>
</Formik>
`;
